code/game/objects/items/devices/miragedrive.dm

```diff
@@ -0,0 +1,70 @@
+/obj/item/mdrive
+	name = "mirage drive"
+	desc = "A device that functions to increase the user's kinetic energy and direct it. Should the user land near other beings, the device will draw from them, slowing them down \
+	and allowing the device to be used sooner. The recharge time for the device scales with the distance traveled, capping out at one minute. If the user's legs are restrained, \
+	they will only be able to jump to a target directly within their vision."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "miragedrive"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	item_state = "mdrive"
+	w_class = WEIGHT_CLASS_SMALL
+	var/cooldown_coefficient = 4 //determines how many deciseconds each tile traveled adds to the cooldown
+	var/access_card = new /obj/item/card/id/captains_spare()
+	var/recharged = TRUE
+	var/cooldown_remaining
```

Comment:
Use time defines where appropriate and use cooldown macros instead of `world.time`
[code\__DEFINES\cooldowns.dm](https://github.com/yogstation13/Yogstation/blob/master/code/__DEFINES/cooldowns.dm#L59-L73)